iothub/device/src/Authentication/HsmAuthentication/HttpClientHelper.cs

```diff
@@ -27,7 +28,7 @@ public static HttpClient GetHttpClient(Uri providerUri)
 
             if (providerUri.Scheme.Equals(UnixScheme, StringComparison.OrdinalIgnoreCase))
             {
-                return new HttpClient(new HttpUdsMessageHandler(providerUri));
+                return new HttpClient(new HttpUdsMessageHandler(providerUri), true);
```

Comment:
Avoid this memory leak.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.-ctor?view=net-7.0#system-net-http-httpclient-ctor(system-net-http-httpmessagehandler-system-boolean)